Integrate gutenberg-mobile release 1.99.0

fluiddot opened the pull request:
## Description
This PR incorporates the 1.99.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5944

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

fluiddot committed:
Release script: Update gutenberg-mobile ref

fluiddot committed:
Update release notes

fluiddot committed:
Update Gutenberg Mobile ref

geriux commented on RELEASE-NOTES.txt:
Like I mentioned in the iOS PR I was wondering if we should mark this entry as internal, what do you think?

geriux reviewed (approved):
LGTM! Friendly reminder to update the Gutenberg mobile commit hash to the release tag. 🚀

fluiddot commented on RELEASE-NOTES.txt:
Yep, good idea. Although it's a crash (i.e. something the user notices), I think the release note doesn't give any value to the user as it's hard to reproduce in the app. I'll update it before merging the PR. Thanks!

fluiddot committed:
Update release notes

Adding `internal` tag to Reanimated crash entry as it's not guaranteed that would fix it.

fluiddot committed:
Update Gutenberg Mobile ref with tag
